add ecr module with example usage

mateuszwu opened the pull request.

mateuszwu committed:
add ecr module with example usage

mateuszwu committed:
fix output name formatting

qbart reviewed (changes requested):
I think we need separation between pull and push permissions.
CI will pull & push, but the server will only pull

mateuszwu committed:
split permissions to push and pull

mateuszwu committed:
rename 'ecr' module to 'ecr-pull-push'

mateuszwu commented:
> I think we need separation between pull and push permissions. CI will pull & push, but the server will only pull

I've split permissions to pull & push.
Additionally, I've changed the module name to `ecr-pull-push` to follow the convention suggested in here https://github.com/Selleo/terraform-aws-iam/pull/2#pullrequestreview-840999677
